src/frontend/app/components/BrowseScreenSearchWidget.tsx

```diff
@@ -106,43 +116,37 @@ const SearchResults = ({
       }
 
       onKeyphraseChange({ keyphrase: value });
-
-      // Set the input value manually when user clicks on a keyword suggestion
-      if (isSuggestion) {
-        const searchInput = document.getElementById('search-input') as HTMLInputElement;
-        searchInput.value = value;
-      }
     },
     [onKeyphraseChange, router, pathname]
   );
 
-  const onKeyphraseChangeDebounced = debounce((value: string) => handleKeyphraseChange(value), 500);
-
   const autocompleteSuggestion = useMemo(() => {
     return !!suggestions.length && suggestions?.[0].text;
   }, [suggestions]);
 
   const handleAutocomplete = useCallback(() => {
-    const searchInput = document.getElementById('search-input') as HTMLInputElement;
-    searchInput.value = !!autocompleteSuggestion ? autocompleteSuggestion : searchInput.value;
-  }, [autocompleteSuggestion]);
+    if (!!autocompleteSuggestion) {
+      setInputValue(autocompleteSuggestion);
```

Comment:
This line seems unnecessary because the same action happens inside handleKeyphraseChange().